jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/tx/DisposeSessionTransactionSynchronization.java

```diff
@@ -42,7 +45,7 @@ public void afterCompletion(int status) {
 	    try {
 	        manager.disposeRuntimeEngine(runtime);
 	    } catch (Throwable e) {
-	        // catch exception as it's only clean up and should not affect runtime
+	        logger.warn("Could not dispose runtime engine", e);
```

Comment:
but isn't this a serious issue? for singleton strategy disposing is a no-op, but both per_request and per_process are using ThreadLocals to cache the runtimeEngine(s). couldnt a failure in disposal lead to stale runtime engines being used in a thread and heap issues?